.github/workflows/checks.yml

```diff
@@ -169,6 +169,13 @@ jobs:
       fail-fast: false
       matrix:
         python-version: [ "3.8", "3.9", "3.10", "3.11" ]
+    env:
+      INTEGRATION_NODE_URL: ${{ secrets.INTEGRATION_NODE_URL }}
+      TESTNET_NODE_URL: ${{ secrets.TESTNET_NODE_URL }}
+      INTEGRATION_ACCOUNT_PRIVATE_KEY: ${{ secrets.INTEGRATION_ACCOUNT_PRIVATE_KEY }}
+      INTEGRATION_ACCOUNT_ADDRESS: ${{ secrets.INTEGRATION_ACCOUNT_ADDRESS }}
+      TESTNET_ACCOUNT_PRIVATE_KEY: ${{ secrets.TESTNET_ACCOUNT_PRIVATE_KEY }}
+      TESTNET_ACCOUNT_ADDRESS: ${{ secrets.TESTNET_ACCOUNT_ADDRESS }}
```

Comment:
any way we could remove the duplication in each job?

Reply:
Probably after refactoring the logic behind those envs it will be possible